packages/blade/src/components/Input/TextArea/__tests__/TextArea.native.test.tsx

```diff
@@ -0,0 +1,130 @@
+import { fireEvent } from '@testing-library/react-native';
+import type { ReactElement } from 'react';
+import React from 'react';
+
+import { TextArea } from '..';
+import renderWithTheme from '~src/_helpers/testing/renderWithTheme.native';
+
+describe('<TextArea />', () => {
+  it('should render', () => {
+    const { toJSON } = renderWithTheme(<TextArea label="Enter name" />);
+
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should display success validation state', () => {
+    const { getByText } = renderWithTheme(
+      <TextArea
+        label="Enter name"
+        validationState="success"
+        successText="Success"
+        helpText="Help"
+        errorText="Error"
+      />,
+    );
+
+    const successText = getByText('Success');
+
+    expect(successText).toBeTruthy();
+  });
+
+  it('should display error validation state', () => {
+    const { getByText } = renderWithTheme(
+      <TextArea
+        label="Enter name"
+        validationState="error"
+        errorText="Error"
+        successText="Success"
+        helpText="Help"
+      />,
+    );
+
+    const errorText = getByText('Error');
+
+    expect(errorText).toBeTruthy();
+  });
+
+  it('should be focussed when autoFocus flag is passed', () => {
+    const placeholder = 'First last';
+    const { getByPlaceholderText } = renderWithTheme(
+      // eslint-disable-next-line jsx-a11y/no-autofocus
+      <TextArea label="Enter name" placeholder={placeholder} autoFocus />,
+    );
+
+    const input = getByPlaceholderText(placeholder);
+
+    // we assume auto focus is working with this prop in place, no simple way of asserting on focus otherwise
+    // @ts-expect-error TS typings not being picked from library
+    expect(input).toHaveProp('autoFocus', true);
+  });
+
+  it('should be disabled when isDisabled flag is passed', () => {
+    const placeholder = 'First last';
+    const { getByPlaceholderText } = renderWithTheme(
+      <TextArea label="Enter name" placeholder={placeholder} isDisabled />,
+    );
+
+    const input = getByPlaceholderText(placeholder);
+    expect(input).toBeDisabled();
+  });
+
+  it('should handle onChange', () => {
+    const placeholder = 'First Last';
+    const onChange = jest.fn();
+    const userName = 'Divyanshu';
+
+    const { getByPlaceholderText } = renderWithTheme(
```

Comment:
can we update the label on react native to have a11y prop so we can do `getByLabel`?

Reply:
Need to fix this #668